### PR TITLE
Search History Implementation

### PR DESCRIPTION
**Pull request by N4J13**

### CODE REVIEW

File: lib/bloc/bus_bloc.dart [hunk 1]
```diff
@@ -17,6 +18,8 @@ class BusBloc extends Cubit<BusState> {
   final formKey = GlobalKey<FormState>();
   List<RouteData> routesData = [];
   List<VehiclesData> vehiclesData = [];
+  final HiveService hiveService = HiveService();
+  List list = [];
```
Comment (N4J13) [inline lib/bloc/bus_bloc.dart]: Change variable name list to searchRecords

File: lib/models/search_record_model.dart [hunk 2]
```diff
@@ -3,15 +3,12 @@ part 'search_record_model.g.dart';
 
 @HiveType(typeId: 0)
 class SearchRecordModel {
-  @HiveField(0)
-  final int? id;
   @HiveField(1)
-  final String? origin;
+  final String? departure;
```
Comment (N4J13) [inline lib/models/search_record_model.dart:7]: Nice

File: lib/presentation/widgets/recent_search.dart [hunk 3]
```diff
@@ -1,14 +1,91 @@
...
-        Text('Recent Search'),
-      ],
+    final busBloc = BlocProvider.of<BusBloc>(context)..getRecentSearch();
+    final double screenWidth = MediaQuery.of(context).size.width;
+    final double screenHeight = MediaQuery.of(context).size.height;
+
+    return BlocConsumer<BusBloc, BusState>(
+      listener: (context, state) {},
+      builder: (context, state) {
+        return SingleChildScrollView(
```
Comment (N4J13) [inline lib/presentation/widgets/recent_search.dart]: SingleChildScrollViewNotNeeded

File: lib/presentation/widgets/recent_search.dart [hunk 4]
```diff
@@ -1,14 +1,91 @@
...
+                          'Clear All',
+                          style: TextStyle(
+                            color: Colors.red,
+                          ),
+                        ),
+                      ),
+                    ],
+                  ),
+                  SizedBox(
+                    height: screenHeight,
```
Comment (N4J13) [inline lib/presentation/widgets/recent_search.dart]: height should not be equal to screenHeight but the total height of  the children its having

**Review by N4J13 — COMMENTED** [review]
.

File: lib/bloc/bus_bloc.dart (hunk 1 above)
Comment (Nishmakv) [reply lib/bloc/bus_bloc.dart]: Done

File: lib/presentation/widgets/recent_search.dart (hunk 3 above)
Comment (Nishmakv) [reply lib/presentation/widgets/recent_search.dart]: Done

File: lib/presentation/widgets/recent_search.dart (hunk 4 above)
Comment (Nishmakv) [reply lib/presentation/widgets/recent_search.dart]: Done

**Pull request merged**